source/main/commands/file-path-find.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        FilePathFind command
+ * CVM-Role:        <none>
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This command finds the absolute path of a file
+ *
+ * END HEADER
+ */
+
+import ZettlrCommand from './zettlr-command'
+import { filetypes as FILETYPES } from '../../common/data.json'
+
+export default class FilePathFind extends ZettlrCommand {
+  constructor (app: any) {
+    super(app, ['file-path-find'])
+  }
+
+  /**
+      * Removes a file.
+      * @param {String} evt The event name
+      * @param  {Object} arg the parameters of the file to be deleted
+      * @return {Boolean} Whether the file was successfully deleted.
+      */
+  async run (evt: string, arg: any): Promise<String> {
+    // Initialise the file as the result of findExact failing
+    let file
+    // It might be linked by ID
+    file = this._app.getFileSystem().findExact((arg as string), 'id')
+    if (file !== undefined) {
+      return file.path
+    }
+    // It's not an ID, so search each type of file
+    for (let type of FILETYPES) {
+      file = this._app.getFileSystem().findExact((arg as string) + type, 'name')
+      if (file !== undefined) {
+        // If we find it, then return it
+        return file.path
```

Comment:
You know what I just realised? Up in the editor hook you are running two successive asynchronous IPC calls to first retrieve the file path and THEN the note's contents. However, why don't you combine these calls, and instead of `return file.path` use `return await this._app.getFileSystem().getFileContents(file)`? Then you have everything you need, skip the second IPC call altogether and can safely refactor the code in the callback into its own function.

Reply:
I agree. I will change the 'file-path-find' to achieve this. Probably renaming it for its new functionality. Would you like me to keep an old version of the file-path-find around so that other people have a command to just grab the absolute path of a file?

Reply:
Nah, I don't think we should do that. I've learned the hard way that you should only implement something as soon as you need it, and not in good faith that maybe someone uses it some day, so it's safe to just overwrite that command! (After all, it's super simple to implement that if we really need it)